src/tests/unit_tests/apis/resource/resource_api_test.py

```diff
@@ -1,6 +1,389 @@
+            ).thenReturn(None)
+        elif mime_type is MimeType.JSON_LD and cause == "not_rdf_graph":
+            when(util.ld_util).get_lod_resource_from_rdf_store(
+                DUMMY_URL,
+                application_settings.get("SPARQL_ENDPOINT"),
+                application_settings.get("URI_NISV_ORGANISATION"),
+            ).thenReturn(None)
+            when(DUMMY_GRAPH).serialize(
+                format=mime_type.to_ld_format(), auto_compact=True
+            ).thenReturn(
+                None
+            )  # shouldn't be called, but we want to check this
+        elif mime_type is MimeType.JSON_LD and cause == "no_serialized_graph":
+            when(util.ld_util).get_lod_resource_from_rdf_store(
+                DUMMY_URL,
+                application_settings.get("SPARQL_ENDPOINT"),
+                application_settings.get("URI_NISV_ORGANISATION"),
+            ).thenReturn(DUMMY_GRAPH)
+            when(DUMMY_GRAPH).serialize(
+                format=mime_type.to_ld_format(), auto_compact=True
+            ).thenReturn(None)
+
+        response = generic_client.get(
+            "offline",
+            resource_query_url(CAT_TYPE, DUMMY_IDENTIFIER),
+            headers={"Accept": mime_type.value},
+        )
+
+        verify(util.ld_util, times=1).generate_lod_resource_uri(
+            ResourceApiUriLevel(CAT_TYPE),
+            DUMMY_IDENTIFIER,
+            application_settings.get("BENG_DATA_DOMAIN"),
+        )
+
+        assert response.status_code == 500
+
+        if mime_type is MimeType.HTML:
+            verify(ResourceAPI, times=1)._get_lod_view_resource(
+                DUMMY_URL,
+                application_settings.get("SPARQL_ENDPOINT"),
+                application_settings.get("URI_NISV_ORGANISATION"),
+            )
+            if cause == "no_html_page":
+                assert "Could not generate an HTML view for this resource" in str(
+                    response.text
+                )
+        else:
+            verify(util.ld_util, times=1).get_lod_resource_from_rdf_store(
+                DUMMY_URL,
+                application_settings.get("SPARQL_ENDPOINT"),
+                application_settings.get("URI_NISV_ORGANISATION"),
+            )
+            verify(DUMMY_GRAPH, times=0 if cause == "not_rdf_graph" else 1).serialize(
+                format=mime_type.to_ld_format(), auto_compact=True
+            )
+            if cause == "not_rdf_graph":
+                assert "No graph created" in str(response.text)
+            elif cause == "no_serialized_graph":
+                assert "Serialisation failed" in str(response.text)
+    finally:
+        unstub()
+
+
+@pytest.mark.parametrize("item_type", [None, "scene", "program", "season", "series"])
+def test__get_lod_view_resource(
+    item_type,
+    application,
+    application_settings,
+    i_scene_graph,
+    i_program_graph,
+    i_season_graph,
+    i_series_graph,
```

Comment:
removed

Reply:
oh no, my mistake. They are used, but just don't show as such because they are loaded in the code depending on the item type:
```

        if item_type:
            test_graph = locals()[f"i_{item_type}_graph"]
```
I've put them back and added a comment